client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala

```diff
@@ -69,7 +70,31 @@ private[sharing] class DeltaSharingFileSystem extends FileSystem with Logging {
       val proxy = new HttpHost(proxyConfig.host, proxyConfig.port)
       clientBuilder.setProxy(proxy)
 
-      if (proxyConfig.noProxyHosts.nonEmpty) {
+      val neverUseHttps = getConf.getBoolean("spark.delta.sharing.never.use.https", false)
+      if (neverUseHttps) {
+        val httpRequestDowngradeExecutor = new HttpRequestExecutor {
+          override def execute(
+                                request: HttpRequest,
+                                connection: HttpClientConnection,
+                                context: HttpContext): HttpResponse = {
+            try {
+              val modifiedUri: URI = {
+                new URIBuilder(request.getRequestLine.getUri).setScheme("http").build()
+              }
+              val wrappedRequest = new RequestWrapper(request)
+              wrappedRequest.setURI(modifiedUri)
+
+              return super.execute(wrappedRequest, connection, context)
+            } catch {
+              case e: Exception =>
+                logInfo("Failed to downgrade the request to http", e)
+            }
+            super.execute(request, connection, context)
+          }
+        }
+        clientBuilder.setRequestExecutor(httpRequestDowngradeExecutor)
+      }
```

Comment:
we allow the `neverUseHttps` case to go into the `noProxyHosts` route planner so that requests go through the path, 
```
 new HttpRoute(target, proxy)

```
where the default `secure` (see Apache docs [here](https://www.javadoc.io/doc/org.apache.httpcomponents/httpclient/4.3.3/org/apache/http/conn/routing/HttpRoute.html#HttpRoute(org.apache.http.HttpHost,%20java.net.InetAddress,%20boolean))) value is false. This prevents the tunneling (`CONNECT`) on https requests.